src/Evolution/Executables/Burgers/EvolveBurgers.hpp

```diff
@@ -62,17 +64,29 @@ struct EvolutionMetavars {
                  StepChoosers::Register::Constant,
                  StepChoosers::Register::Increase>;
 
+  using compute_rhs = tmpl::list<
+      Actions::ComputeVolumeFluxes,
+      dg::Actions::SendDataForFluxes<EvolutionMetavars>,
+      Actions::ComputeVolumeDuDt,
+      dg::Actions::ReceiveDataForFluxes<EvolutionMetavars>,
+      // dg::Actions::ApplyBoundaryFluxesGlobalTimeStepping,  // for GTS
+      Actions::RecordTimeStepperData>;
+  using update_variables = tmpl::list<
+      dg::Actions::ApplyBoundaryFluxesLocalTimeStepping,  // for LTS
+      Actions::UpdateU>;
+
+  struct EvolvePhaseStart;
   using component_list = tmpl::list<DgElementArray<
       EvolutionMetavars,
-      tmpl::list<Actions::AdvanceTime, Actions::FinalTime,
-                 Actions::ChangeStepSize<step_choosers>,
-                 Actions::ComputeVolumeFluxes,
-                 dg::Actions::SendDataForFluxes<EvolutionMetavars>,
-                 Actions::ComputeVolumeDuDt,
-                 dg::Actions::ReceiveDataForFluxes<EvolutionMetavars>,
-                 Actions::RecordTimeStepperData,
-                 dg::Actions::ApplyBoundaryFluxesLocalTimeStepping,
-                 Actions::UpdateU>>>;
+      tmpl::flatten<tmpl::list<
+          SelfStart::self_start_procedure<compute_rhs, update_variables>,
+          Actions::Label<EvolvePhaseStart>,
+          Actions::AdvanceTime,
+          Actions::FinalTime,
+          Actions::ChangeStepSize<step_choosers>,
+          compute_rhs,
+          update_variables,
+          Actions::Goto<EvolvePhaseStart>>>>>;
```

Comment:
This Label/Goto mechanism basically bypasses the algorithm architecture, having an initialization and action loop _inside_ the actual action loop, which is only executed once here I think. Can we use use a **phase** for the self-starting instead?

Reply:
That would be the correct way to do it, but phases aren't fully implemented yet.

Reply:
Is the GOTO mechanism only a temporary hack to get this working then? I am a little scared to have that mechanism in the code.. 😄

Reply:
Expanding on what @wthrowe said: phases in the future will have an action list for them. That is, the action list is different for each phase because we want different algorithms in each phase. We originally only expected to need this for AMR (maybe, but now much more clearly so), and when we transition from merger to ringdown. That would've bought us something like at least 6 months time to implement them. This will be one of the first things implemented in December.

Reply:
Ok, so will we remove the Label/Goto functionality in that context? Because I think it can be misused and hard to debug.

Reply:
Yes